docs/services/ml/upload-training-script.md

```diff
@@ -10,48 +10,231 @@ no_service: true
+               json_line = json.loads(line)
+               image_filenames.append(json_line["image_path"])
+
+               annotations = json_line["classification_annotations"]
+               labels = [unknown_label]
+               for annotation in annotations:
+                   if model_type == multi_label:
+                       if annotation["annotation_label"] in all_labels:
+                           labels.append(annotation["annotation_label"])
+                   if model_type == single_label:
+                       if annotation["annotation_label"] in all_labels:
+                           labels = [annotation["annotation_label"]]
+               image_labels.append(labels)
+       return image_filenames, image_labels
+
+
+   # Parse the dataset file (produced and stored in Viam) to get
+   # bounding box annotations
+   # Used for training object detection models
+   def parse_filenames_and_bboxes_from_json(
+       filename: str,
+       all_labels: ty.List[str],
+   ) -> ty.Tuple[ty.List[str], ty.List[str], ty.List[ty.List[float]]]:
+       """Load and parse JSON file to return image filenames
+       and corresponding labels with bboxes.
+       Args:
+           filename: JSONLines file containing filenames and bboxes
+       """
+       image_filenames = []
+       bbox_labels = []
+       bbox_coords = []
+
+       with open(filename, "rb") as f:
+           for line in f:
+               json_line = json.loads(line)
+               image_filenames.append(json_line["image_path"])
+               annotations = json_line["bounding_box_annotations"]
+               labels = []
+               coords = []
+               for annotation in annotations:
+                   if annotation["annotation_label"] in all_labels:
+                       labels.append(annotation["annotation_label"])
+                       # Store coordinates in rel_yxyx format so that
+                       # we can use the keras_cv function
+                       coords.append(
+                           [
+                               annotation["y_min_normalized"],
+                               annotation["x_min_normalized"],
+                               annotation["y_max_normalized"],
+                               annotation["x_max_normalized"],
+                           ]
+                       )
+               bbox_labels.append(labels)
+               bbox_coords.append(coords)
+       return image_filenames, bbox_labels, bbox_coords
+
+
+   # Build the model
+   def build_and_compile_model(
+       labels: ty.List[str], model_type: str, input_shape: ty.Tuple[int, int, int]
+   ) -> Model:
+       """Builds and compiles a model
+       Args:
+           labels: list of string lists, where each string list contains up to N_LABEL labels associated with an image
+           model_type: string single_label or multi_label
+           input_shape: 3D shape of input
+       """
+
+       # TODO: Add logic to build and compile model
```

Comment:
Is this supposed to be a TODO? If so, I do think it would be helpful to have a working example, but I will defer to you

Reply:
What would a working example look like?

Reply:
[This](https://github.com/viamrobotics/docs/pull/3232/files#diff-3594b00458ec93c1fa54469b947146d47f304bedf08dbc11ec2f677e58ca13d1L212-L261) is what a working example would look like.

Reply:
Yes because it's a template. I could put an example with the explanation but this code relies on other methods so it's not that simple. And it would likely get outdated and not get updated

Reply:
I've added the examples